### DISCUSSION TITLE
Small amounts of heterogeneous data significantly impact the prediction results for lithium metal systems.

### DISCUSSION THREAD

**Discussion by ikuki-ikuki** [deepmodeling/deepmd-kit#4788] [Q&A]
Hello everyone, I'm encountering perplexing issues while using DeepMD-kit for my research and would appreciate your insights. My study aims to simulate lithium dendrite growth in batteries by stacking lithium atoms onto a ternary layered material (effectively making it a quaternary system). I began by generating 20,000 initial data points through VASP calculations on bulk and slab models of lithium's three polymorphs (BCC, HCP, FCC), applying various perturbations. A potential trained solely on this lithium dataset accurately described metallic lithium behavior and successfully captured dendrite formation. However, when I augmented the training set with just 960 additional configurations of the substrate material (bulk  structures), the potential catastrophically failed to describe lithium metal - showing drastically increased energy/force errors and exhibiting unphysical thermal motion in LAMMPS simulations.（Fig 1.) What underlying mechanisms could cause this degradation?
Attaching the dp test results, dp train script content.

![Figure_1](https://github.com/user-attachments/assets/bd36ba20-a367-4658-a2ca-d14ac1996b3f)

`{

    "_comment": "that's all",
    "model": {
        "type_map": [
            "Ti",
            "In",
            "B",
            "Li"
        ],
        "descriptor": {
            "type": "se_e2_a",
            "sel": "auto",
            "rcut_smth": 0.5,
            "rcut": 6.0,
            "neuron": [
                25,
                50,
                100
            ],
            "resnet_dt": false,
            "axis_neuron": 16,
            "seed": 1,
            "_comment": "that's all"
        },
        "fitting_net": {
            "neuron": [
                240,
                240,
                240
            ],
            "resnet_dt": true,
            "seed": 1,
            "_comment": "that's all"
        },
        "_comment": "that's all"
    },
    "learning_rate": {
        "type": "exp",
        "decay_steps": 2500,
        "start_lr": 0.001,
        "stop_lr": 3.51e-08,
        "_comment": "that's all"
    },
    "loss": {
        "type": "ener",
        "start_pref_e": 0.02,
        "limit_pref_e": 1,
        "start_pref_f": 1000,
        "limit_pref_f": 1,
        "start_pref_v": 0,
        "limit_pref_v": 0,
        "_comment": "that's all"
    },
    "training": {
        "training_data": {
            "systems": [
                "..../dp/data/BULK/700/train",
	"..../data/Li/BCC/NVT/350/train",
	"..../data/Li/BCC/NVT/400/train",
        ......
            ],
            "batch_size": "auto",
            "_comment": "that's all"
        },
        "validation_data": {
            "systems": [
                "...../data/BULK/700/val",
	"..../data/Li/BCC/NVT/350/val",
	"..../data/Li/BCC/NVT/400/val",
         ......
            ],
            "batch_size": "auto",
            "numb_btch": 1,
            "_comment": "that's all"
        },
        "numb_steps": 1000000,
        "seed": 10,
        "disp_file": "lcurve.out",
        "disp_freq": 1000,
        "save_freq": 10000,
        "_comment": "that's all"
    }
}

**Comment by wanghan-iapcm** [comment]
have you used exactly the same vasp settings for the initial and augmentation datasets? 

what is the training and validation accuracy for the initial and augmentation datasets?

**Reply by ikuki-ikuki** [reply]
Thanks for your reply，
I dont understand the meaning of "accuracy for the initial and augmentation".All my AIMD datasets are training under the same settings by vasp5.4.4. Here are my specific script parameter settings.
ISTART =   0  
LWAVE  =   .FALSE.
LCHARG =   .FALSE.
ENCUT  =   500
NELM   =   200
EDIFF  =   1E-4
NSW    =   800
NBLOCK =   1   
KBLOCK =   1
ISYM   =   0
ISIF   =   2
POTIM  =   3
TEBEG  =   350
SMASS  =   0
MDALGO =   2
ISMEAR =   1
SIGMA  =   0.2
IALGO  =   48
ALGO   =   N
LREAL  =   T
IBRION =   0
NCORE = 8
LSCALAPACK = .FALSE.
All parameters except simulation temperature(TEBEG) and step size(POTIM) are consistent. I've verified that all results demonstrate final convergence.
Additionally, I have actually conducted more AIMD calculations, including:

- ​​17,528 entries​​ for bulk/slab/perturbed substrate materials
- ​​17,106 entries​​ for Li-substrate interfaces
- ​​18,895 entries​​ for pristine Li systems

All data subsets successfully described their respective systems ​​when trained separately for specialized potentials​​. However, when integrated into a unified potential function, ​​the combined dataset fails to accurately describe any system​​ – even previously trained models become unusable for MD simulations.
Could this be related to the training steps or learning rate settings in my training process?

**Reply by TypicalBarmalei** [reply]
> Thanks for your reply， I dont understand the meaning of "accuracy for the initial and augmentation".All my AIMD datasets are training under the same settings by vasp5.4.4. Here are my specific script parameter settings. ISTART = 0 LWAVE = .FALSE. LCHARG = .FALSE. ENCUT = 500 NELM = 200 EDIFF = 1E-4 NSW = 800 NBLOCK = 1 KBLOCK = 1 ISYM = 0 ISIF = 2 POTIM = 3 TEBEG = 350 SMASS = 0 MDALGO = 2 ISMEAR = 1 SIGMA = 0.2 IALGO = 48 ALGO = N LREAL = T IBRION = 0 NCORE = 8 LSCALAPACK = .FALSE. All parameters except simulation temperature(TEBEG) and step size(POTIM) are consistent. I've verified that all results demonstrate final convergence. Additionally, I have actually conducted more AIMD calculations, including:
> 
>     * ​​17,528 entries​​ for bulk/slab/perturbed substrate materials
> 
>     * ​​17,106 entries​​ for Li-substrate interfaces
> 
>     * ​​18,895 entries​​ for pristine Li systems
> 
> 
> All data subsets successfully described their respective systems ​​when trained separately for specialized potentials​​. However, when integrated into a unified potential function, ​​the combined dataset fails to accurately describe any system​​ – even previously trained models become unusable for MD simulations. Could this be related to the training steps or learning rate settings in my training process?

Hopefully you have resolved your issue, but if not, I'll share my thoughts on that (even though I'm not that experienced with DeePMD itself). 
As @wanghan-iapcm mentioned, it's likely the inconsistency in data accuracy of your dataset, as your accuracy drops when "**merging**" datasets.
1. As your "main" element is Li, VASP highly recommends using the Li_sv PP with ENMAX of 499 eV - really close to your set one of 500 eV. It may cause problems even in equilibrium structures and very possible will impact structures away from it (which, obviously, will appear if you want your potential to cover structures away from full equilibrium). It's a good idea to take a couple of systems, just a few, and check results consistency with respect to ENCUT and KSPACING (or your manual KPOINTS mesh). I think stress and force values may differ significantly. You mention: I_'ve verified that all results demonstrate final convergence._, but I'm not sure what exactly you mean.
2. EDIFF = 1E-4 may be too low. It shouldn't lead to the behavior you describe, but it's still a good idea to get rid of "noisiness" in energy & force values.
